src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -909,6 +840,68 @@ private static DetailNode getSinceJavadocTag(DetailNode javadoc) {
         return javadocTagWithSince;
     }
 
+    /**
+     * Returns {@code true} if {@code actualVersion} ≥ {@code requiredVersion}.
+     * Both versions have any trailing "-SNAPSHOT" stripped before comparison.
+     *
+     * @param actualVersion   e.g. "8.3" or "8.3-SNAPSHOT"
+     * @param requiredVersion e.g. "8.3"
+     * @return {@code true} if actualVersion exists, and, numerically, is at least requiredVersion
+     */
+    private static boolean isVersionAtLeast(String actualVersion,
+                                            String requiredVersion) {
+        boolean result = false;
+
+        final String actClean = actualVersion
+                .replace(SNAPSHOT_SUFFIX, "")
+                .trim();
+        final String reqClean = requiredVersion
+                .replace(SNAPSHOT_SUFFIX, "")
+                .trim();
+
+        try {
+            final Version vAct = Version.parse(actClean);
+            final Version vReq = Version.parse(reqClean);
+            result = vAct.compareTo(vReq) >= 0;
+        }
+        catch (IllegalArgumentException iae) {
+            // Fallback: compare each numeric segment
+            final Pattern compiledDotRegexPattern = Pattern.compile(DOT_REGEX);
+
+            final String[] partsAct = compiledDotRegexPattern.split(actClean);
+            final String[] partsReq = compiledDotRegexPattern.split(reqClean);
+            final int length = Math.max(partsAct.length, partsReq.length);
+
+            boolean compared = false;
+            for (int index = 0; index < length; index++) {
+                final int a;
+                if (index < partsAct.length) {
+                    a = Integer.parseInt(partsAct[index]);
+                }
+                else {
+                    a = 0;
+                }
+                final int r;
```

Comment:
Same as above


```suggestion
                final int required;
```

Or



```suggestion
                final int requiredVersion;
```